caas/kubernetes/provider/k8s.go

```diff
@@ -246,22 +243,29 @@ func (k *kubernetesClient) ensureNamespaceAnnotationForControllerUUID(controller
 	if err != nil && !errors.IsNotFound(err) {
 		return errors.Trace(err)
 	}
+	// If the namespace doesn't exist then there is nothing we can do.
+	if errors.IsNotFound(err) {
+		return nil
+	}
 	annotationControllerUUIDKey := utils.AnnotationControllerUUIDKey(isLegacy)
+
+	// If juju doesn't own the namespace then we should leave it alone.
+	if ns != nil && !isK8sObjectOwnedByJuju(ns.ObjectMeta) {
+		return nil
```

Comment:
This seems like a problem - juju will now be expecting the namespace to have the annotations when we are lying and saying it does when it doesn't. So later operations like destroy controller etc will get confused. Is there a reason why we don't error here?